lib/Sema/CSDiag.cpp

```diff
@@ -2743,11 +2743,24 @@ diagnoseUnviableLookupResults(MemberLookupResult &result, Type baseObjTy,
           return;
         }
       }
-        
+
+      // Check whether the instance member is declared on parent context and if so
+      // provide more specialized message
+      auto memberTypeContext = member->getDeclContext()->getInnermostTypeContext();
+      auto currentTypeContext = CS.DC->getInnermostTypeContext();
+      if (memberTypeContext->getSyntacticDepth() < currentTypeContext->getSyntacticDepth()) {
```

Comment:
Syntactic depth will catch this for explicit nesting, but what about extensions of nested types?

Reply:
Good catch! I will check it, thank you!

Reply:
I've checked it (sorry for the delay). In case of extensions it doesn't even go to this path of code. In case of writing code like:
```
struct Outer {
  var outer: Int

  struct Inner {
    var inner: Int
  }
}

extension Outer.Inner {
    func sum() -> Int {
      return inner + outer
    }
}
```
message is `use of unresolved identifier 'outer'`. Nevertheless, I've changed check to use `SemanticDepth` instead. Thanks!